ankaios_sdk/ankaios.py

```diff
@@ -0,0 +1,733 @@
+        """
+        raise NotImplementedError(
+            "set_config_from_file is not implemented yet."
+            )
+
+    def set_config(self, name: str, config: Union[dict, list, str]) -> bool:
+        """
+        Set the config with the provided name.
+        If the config exists, it will eb replaced.
+
+        Args:
+            name (str): The name of the config.
+            config (Union[dict, list, str]): The config dictionary.
+
+        Returns:
+            bool: True if the config was set successfully, False otherwise.
+        """
+        raise NotImplementedError("set_config is not implemented yet.")
+
+    def get_configs(self) -> dict:
+        """
+        Get the configs. The keys will be the names.
+
+        Returns:
+            dict: The configs dictionary.
+        """
+        raise NotImplementedError("get_configs is not implemented yet.")
+
+    def get_config(self, name: str) -> Union[dict, list, str]:
+        """
+        Get the config with the provided name.
+
+        Args:
+            name (str): The name of the config.
+
+        Returns:
+            Union[dict, list, str]: The config.
+        """
+        raise NotImplementedError("get_config is not implemented yet.")
+
+    def delete_configs(self) -> bool:
+        """
+        Delete all the configs.
+
+        Returns:
+            bool: True if the configs were deleted successfully,
+                False otherwise.
+        """
+        raise NotImplementedError("delete_configs is not implemented yet.")
+
+    def delete_config(self, name: str) -> bool:
+        """
+        Delete the config.
+
+        Args:
+            name (str): The name of the config.
+
+        Returns:
+            bool: True if the config was deleted successfully, False otherwise.
+        """
+        raise NotImplementedError("delete_config is not implemented yet.")
+
+    def get_state(self, timeout: float = DEFAULT_TIMEOUT,
+                  field_masks: list[str] = None) -> CompleteState:
+        """
+        Send a request to get the complete state.
+
+        Args:
+            timeout (float): The maximum time to wait for the response,
+                in seconds.
+            field_masks (list[str]): The list of field masks to filter
+                the state.
+
+        Returns:
+            CompleteState: The complete state object.
+        """
+        request = Request(request_type="get_state")
+        if field_masks is not None:
+            request.set_masks(field_masks)
+        try:
+            response = self._send_request(request, timeout)
+        except TimeoutError as e:
+            self.logger.error("%s", e)
+            return None
+
+        # Interpret response
+        (content_type, content) = response.get_content()
+        if content_type == "error":
+            self.logger.error("Error while trying to get the state: %s",
+                              content)
+            return None
```

Comment:
Thinking of the Rust sdk because the errors are handled differently there (in that case, the method would return a Result), I should do something similar in python. Currently it's similar to a `Option`, that the rust supports as well, because it's easier. I could throw the error as well and let the user catch it.

Your idea would be to create a list of Errors for the sdk that the user catches and use them instead of the default python errors right?

Reply:
Done.